lib/instrumentation/core/http2.js

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const NAMES = require('../../metrics/names')
+const recordExternal = require('../../metrics/recorders/http_external')
+const logger = require('../../logger').child({ component: 'http2' })
+const synthetics = require('../../synthetics')
+const urltils = require('../../util/urltils.js')
+const cat = require('#agentlib/util/cat.js')
+module.exports = initialize
+
+function initialize(agent, http2, moduleName, shim) {
+  shim.wrap(http2, 'connect', wrapConnect)
+  shim.wrapReturn(http2, 'connect', wrapSession)
+  const moduleCtx = this
+
+  /**
+   * Wraps the http2 connect method in instrumentation, and saves the connect URL.
+   * The .request() method's argument doesn't retain the protocol of the connection,
+   * but it's available here as arg[0]
+   *
+   * @param {object} shim the generic shim with which the agent instruments
+   * @param {Function} fn the original connect function
+   * @returns {Function} an instrumented connect function
+   */
+  function wrapConnect(shim, fn) {
+    return function wrappedConnect(...args) {
+      moduleCtx.http2ConnectUrl = args[0]
+      return fn.apply(this, args)
+    }
+  }
+
+  /**
+   * Wraps the ClientHttp2Session returned by http2.connect()
+   *
+   * @param {object} shim the generic shim with which the agent instruments
+   * @param {Function} fn the original connect function
+   * @param {String} fnName name of the connect function
+   * @param {EventEmitter} session the newly created ClientHttp2Session
+   */
+  function wrapSession(shim, fn, fnName, session) {
+    shim.wrap(session, ['request'], wrapRequest)
```

Comment:
there's no reason to wrap and wrapReturn, you can just wrap, do what you need before, call the original, apply the instrumentation on result and return accordingly